pkg/asset/machines/master.go

```diff
@@ -147,7 +147,7 @@ func (m *Master) Generate(dependencies asset.Parents) error {
 
 	ic := installConfig.Config
 
-	pool := ic.ControlPlane
+	pool := ic.ControlPlane.Copy()
```

Comment:
I think we can get away with doing a shallow copy. None of the non-platform fields are changed and the platform fields are overwritten.
```suggestion
	pool := *ic.ControlPlane
```